File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      String.ends_with?(input, "?") -> "Sure."
      String.trim(input) == "" -> "Fine. Be that way!"
      String.upcase(input) == input and Regex.match?(~r/[[:alpha:]]+/, input) -> "Whoa, chill out!"
      true -> "Whatever."
    end
  end
end

File: clojure/bob/src/bob.clj
(ns bob)
(require '[clojure.string :as str])

(defn response-for [phrase]
  (cond
    (and (= (str/upper-case phrase) phrase)
         (re-seq #"[A-Za-z]" phrase)) "Whoa, chill out!"
    (str/ends-with? phrase "?") "Sure."
    (str/blank? (str/trim phrase)) "Fine. Be that way!"
    :else "Whatever."))
